core/arch/riscv/kernel/boot.c

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 NXP
+ */
+
+#include <assert.h>
+#include <compiler.h>
+#include <config.h>
+#include <console.h>
+#include <keep.h>
+#include <kernel/boot.h>
+#include <kernel/linker.h>
+#include <kernel/misc.h>
+#include <kernel/panic.h>
+#include <kernel/thread.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <mm/tee_mm.h>
+#include <mm/tee_pager.h>
+#include <riscv.h>
+#include <sbi.h>
+#include <stdio.h>
+#include <trace.h>
+#include <util.h>
+
+#include <platform_config.h>
+
+#define PADDR_INVALID               ULONG_MAX
+
+unsigned long _start_addr;
+unsigned long _boot_args[4];
+
+uint32_t sem_cpu_sync[CFG_TEE_CORE_NB_CORE];
+
+void init_sec_mon(unsigned long nsec_entry __maybe_unused)
+{
+	assert(nsec_entry == PADDR_INVALID);
+	/* Do nothing as we don't have a secure monitor */
+}
+
+#ifdef CFG_RISCV_S_MODE
+static void start_secondary_cores(void)
+{
+	int ret;
+	uint32_t i;
+	size_t pos = get_core_pos();
+
+	for (i = 0; i < CFG_TEE_CORE_NB_CORE; i++) {
+		if (i != pos) {
+#ifdef CFG_RISCV_SBI
+			ret = sbi_boot_hart(i, _start_addr, (unsigned long)i);
+#else
+			ret = -1;
+#endif /*CFG_RISCV_SBI*/
+			if (ret != 0)
```

Comment:
What's the point with this when `sbi_boot_hart()` isn't called?

Reply:
We will not able to start the secondary hart when we are in S-Mode.